### ISSUE TITLE
API: VM Created from deleted compute offering 

### ISSUE THREAD

**Issue by azmidhun** [apache/cloudstack#6740]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
4.16.1
~~~

##### CONFIGURATION

Compute offerings
##### OS / ENVIRONMENT

CentOS7

##### SUMMARY

VM can be created from a deleted offering via API

##### STEPS TO REPRODUCE

~~~
Create a compute offering and note the UUID
Remove the Compute offering
Make sure its removed from UI
Create a instance using the UUID of the deleted Offering
Instance Get created VIA compute offering
~~~

##### EXPECTED RESULTS

~~~
Instance should not get created via API using deleted offering
~~~

##### ACTUAL RESULTS

~~~
Instance got created VIA deleted compute offering
~~~

**Comment by DaanHoogland** [comment]
@azmidhun you are right this is the case, A workaround to make sure the offering is no longer used you need to set the field `removed` on it. There is no APIU to this at the moment. that would be a good improvement. The issue with that is that no exisitng VM may be using it or problems will arise when actions on those VMs are taken. So take care if you use my workaround.

**Comment by DaanHoogland** [comment]
fixed by #7063
